third_party/terraform/resources/resource_compute_region_instance_group_manager.go.erb

```diff
@@ -299,6 +299,34 @@ func resourceComputeRegionInstanceGroupManager() *schema.Resource {
 					},
 				},
 			},
+			"apply_updates_to_instances": {
```

Comment:
We've been thinking about how to represent imperative APIs like this - I'm not sure it fits well into Terraform, can you give me a short overview of the user's experience of using this block?

If I understand right, I think this could work, but I'd like to be sure I follow.

Reply:
The way you apply updates to instances managed via an IGM is particularly poor. You can set update policy to PROACTIVE and Google at minimum restarts the instance to apply the change of the template. If the template is changing something like network tags or metadata that doesn't need a reboot/replace which is why Google introduced the refresh option. Also, with PROACTIVE there is no way of controlling the maximum disruption an IGM update can do which is annoying. 

Implementing this introduces new options:
- Limit the IGMs most allowed disruption when updating instances you don't want to replace.
- Update fields on managed instances without RECREATE/RESTART for fields can be done such as labels, metadata etc.

Reply:
Any updates? I would like to move forward with it.

Reply:
We discussed this today in our meeting and I don't think we're comfortable supporting imperative actions like this right now.  We do recognize that this is valuable and we're thinking about how to support your use case, which we recognize is valid.

In particular, we are concerned about the way this update works - it triggers unconditionally if any other field is updated, but not on every `apply` (there is no way to do this in Terraform).  We think that's not very intuitive.

Reply:
Couple of thoughts:
- The API call needs to be executed only when the version.instance_template field is changed.
- Compute Engine Product Team should have incorporated this update method as part of the patch method.

I work at a Google Cloud Partner so i'll reach to out to my contact and see if the GCE Product Team can rework their APIs.

Reply:
I have updated the logic to trigger when version block is changed. `version.0.instance_template` won't work as you can specify multiple versions.

Let me know if it is good to go and i'll update the zonal igm to have this feature too.

Reply:
Great!  This looks great.  I have one naming quibble - can you put this block as `update_policy.opportunistic_update_settings` or something like that?